File: program_data/github_java_sort_function/10/82.java
public void bucketSort(int[] data){
        int max = data[0] , min = data[0];
        for (int i = 0; i < data.length; i++) {
            if(max < data[i]){
                max = data[i];
            }
            if(min > data[i]){
                min = data[i];
            }
        }
        
        int bucketSize = (max - min + 1)/10 + 1;
        Node[] bucket = new Node[bucketSize];
        for (int i = 0; i < bucket.length; i++) {
            bucket[i] = new Node();
        }
        
        for (int i = 0; i < data.length; i++) {
            int k = (data[i] - min + 1)/10;
            bucket[k].node[bucket[k].num] = data[i];
            bucket[k].num++;
        }
        
        int pos = 0 ; 
        
        QuickSort qs = new QuickSort();
        for (int i = 0; i < bucket.length; i++) {
            qs.quickSortInt(bucket[i].node, bucket[i].num, 0, bucket[i].num-1);
            
            for (int j = 0; j < bucket[i].num; j++ , pos++) {
                data[pos] = bucket[i].node[j];
            }
        }
        
        
    }

}

File: program_data/PKU_raw/76/82.c
int main()
{
	int a,b,e,i,j,m,n;
	int f[num];
	int g[num];
	m=2;
	a=10000;
	b=0;
	scanf("%d",&n);
	for(i=0;i<n;i++)
	{
		scanf("%d",&(f[i]));
		scanf("%d",&(g[i]));
	}
	for(i=0;i<n;i++)
	{
		if(f[i]<a)
			a=f[i];
		if(g[i]<a)
			a=g[i];
		if(f[i]>b)
			b=f[i];
		if(g[i]>b)
			b=g[i];
	}
	for(j=n-1;j>0;j--)
	{
		for(i=0;i<j;i++)
		{
			if(f[i]>f[i+1])
			{
				e=f[i];
				f[i]=f[i+1];
				f[i+1]=e;
				e=g[i];
				g[i]=g[i+1];
				g[i+1]=e;
			}
		}
	}
	e=0;
	for(i=1;i<n;i++)
	{
		for(j=0;j<i;j++)
		{
			if(f[i]>g[j])
				e++;
		}
		if(i==e)
			{
			m=3;
			break;
		}
		else
			e=0;
	}
	if(m==2)
	printf("%d %d",a,b);
	else
		printf("no");
   return 0;
}